src/System.ServiceModel.Http/src/System/ServiceModel/Channels/HttpResponseMessageHelper.cs

```diff
@@ -144,25 +144,30 @@ private async Task<bool> ValidateContentTypeAsync(TimeoutHelper timeoutHelper)
                 {
                     throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ProtocolException(SR.HttpContentTypeHeaderRequired));
                 }
+
                 return false;
             }
-            else if (_contentLength != 0)
+
+            if (_contentLength <= 0)
```

Comment:
_contentLength will have a value of -1 if the content length header is missing. They could be using chunked encoding, or it could be an HTTP/1.0 response where it closes the connection after the response has been sent. So this isn't correct as it will falsely return that there is no content when that might not be the case. At this point, we know there's a content type provided, so we only want to return false if _contentLength is exactly 0. If it's -1, we don't have enough info. To do this correctly needs quite a bit of refactoring. In the case of a -1 value, we need to attempt to pre-read a single byte from the stream. When there's no actual content (HTTP/1.0 has closed the connection, or it's chunked encoded with just a 0 length chunk), we will get 0 bytes read and know there's no content.